### ISSUE TITLE
[QST] What should be the default delimiters for string splitting functions?

### ISSUE THREAD

**Issue by beckernick** [rapidsai/cudf#3630]
Various string functions involving splitting strings. There are some subtle differences in the default settings across these functions. We should make sure that we're aligned on the proper defaults per function (which may intentionally vary).

E.g., should the carriage return and newline delimiters be considered part of the default whitespace delimiter for tokenize (docs indicate whitespace is the default), but not token count.

```python
import nvtext
import nvstrings
s = 'hel\nlo\rthe re'
s = nvstrings.to_device(s)
print(nvtext.tokenize(s, delimiter=' '))
print(nvtext.tokenize(s))
['hel\nlo\rthe', 're']
['hel', 'lo', 'the', 're']
```

```python
import nvtext
import nvstrings
s = 'hel\nlo\rthe re'
s = nvstrings.to_device(s)
print(nvtext.token_count(s, delimiter=' '))
print(nvtext.token_count(s))
[2]
[2]
```

**Comment by davidwendt** [comment]
This may be related to #2801

**Comment by harrism** [comment]
@beckernick @davidwendt is this discussion still relevant? I see that in #4035 @davidwendt has gone with any white space as the default delimiter.

**Comment by davidwendt** [comment]
I believe this was addressed in #2801 / #3704 and that #4035 will honor the default delimiter as whitespace.
I will leave it up to @beckernick to close this if he is satisfied.
